### PR TITLE
updates suggested during the docs bug bash

### PR DESCRIPTION
**Pull request by BillWagner**
Various formatting and clarity changes.

/cc @tdykstra @mairaw

### CODE REVIEW

**Comment by cartermp** [comment]
I like the small, one-sentence summary under each bullet point.  Really great way to give a hint as to what the content it links to will talk about.

File: docs/csharp/csharp-6.md [hunk 1]
```diff
@@ -17,18 +17,30 @@ ms.assetid: 4d879f69-f889-4d3f-a781-75194e143400
...
+* [Read-only Auto-properties](#read-only-auto-properties):
+    - You can create read-only auto-properties that can be set only in constructors.
+* [Auto-Property Initializers](#auto-property-initializers):
+    - You can write initialization expressions to set the initial value of an auto-property.
+* [Expression Bodied function members](#expression-bodied-function-members):
+    - You can author one-line methods using lambda expressions.
+* [using static](#using-static):
+    - You can import all the methods of a single class into the current namespace.
+* [Null - conditional operators](#null-conditional-operators):
+    - The null-conditional operator enables you to concisely and safely access members of an object while still checking for null.
```
Comment (tdykstra) [inline docs/csharp/csharp-6.md]: Or follow the pattern of the others and say "You can concisely ..."

File: docs/csharp/csharp-6.md [hunk 2]
```diff
@@ -17,18 +17,30 @@ ms.assetid: 4d879f69-f889-4d3f-a781-75194e143400
...
+* [String Interpolation](#string-interpolation):
+    - You can write string formatting expressions using inline expressions instead of positional arguments.
+* [Exception filters](#exception-filters):
+    - You can catch expressions based on properties of the exception or other program state. 
+* [nameof Expressions](#nameof-expressions):
+    - You can let the compiler generate string representations of symbols.
+* [await in catch and finally blocks](#await-in-catch-and-finally-blocks):
+    - You can use `await` expressions in locations that previously disallowed.
```
Comment (tdykstra) [inline docs/csharp/csharp-6.md]: Add "them" at the end?

**Review by tdykstra — APPROVED** [review]
The summaries after the links are a nice addition.

**Review by mairaw — APPROVED** [review]

**Comment by BillWagner** [comment]
closing and opening to force a new build.

**Pull request merged**